docs/npm_nodejs.md

```diff
@@ -0,0 +1,59 @@
+# Using npm/Node.js code
+
+While Deno is pretty powerful itself, there is a large eco-system of code in the
+[npm](https://npmjs.com/) registry, and many people will want to re-leverage
+tools, code and libraries that are built for [Node.js](https://nodejs.org/). In
+this chapter we will explore how to use it.
+
+The good news, is that in many cases, it _just works_.
+
+There are some foundational things to understand about differences between
+Node.js and Deno that can help in understanding that challenges might be faced:
+
+- Current Node.js supports both CommonJS and ES Modules, while Deno only
+  supports ES Modules. The addition of stabilized ES Modules in Node.js is
+  relatively recent and most code written for Node.js is in the CommonJS format.
+- Node.js has quite a few built-in modules that can be imported and they are a
+  fairly expansive set of APIs. Deno has no built-in modules and only provides a
+  single global namespace/variable named `Deno` and it is part of our goals to
+  try to keep the APIs there to a minimum. Lots of code written for Node.js
+  expects/depends upon these built-in APIs to be available.
+- Node.js has a "magical" module resolution algorithm, where you can import
+  bare-specifiers (e.g. `react` or `lodash`) and Node.js will look in your local
+  and global `node_modules` for a path, introspect the `package.json` and try to
+  see if there is a module named the right way. Deno has no "magical"
+  resolution, much in the way a web browser doesn't have "magical" resolution.
+  For local files, Deno expects a full module name, including the extension.
+  When dealing with remote imports, Deno expects the web server to do any
+  "resolving" and provide back the media type of the code (just like a
+  web-browser).
+- Node.js effectively doesn't work without a `package.json` file. Deno doesn't
+  require an external meta-data file to function or resolve modules.
+- Node.js doesn't currently support remote HTTP imports. It expects all 3rd
+  party code to be installed locally on your machine using a package manager
+  like `npm` into the local or global `node_modules` folder. Deno supports
+  remote HTTP imports (as well as `data` and `blob` URLs) and will go ahead and
+  fetch the remote code and cache it locally, similar to the way a browser
+  works.
+
+In order to help mitigate these differences, we will further explore in this
+chapter:
+
+- Using the [`std/node`](./npm_nodejs/std_node.md) modules of the Deno standard
+  library to "polyfill" the built-in modules of Node.js
+- Using [CDNs](./npm_nodejs/cdns.md) to access the vast majority of npm packages
+  in ways the work under Deno.
+- How [import maps](./npm_nodejs/import_maps.md) can be used to provide "bare
+  specifier" imports like Node.js under Deno, without needing to use a package
+  manager to install packages locally.
+- And finally, a general section of [frequently asked
+  questions][./npm_nodejs/faqs.md]
```

Comment:
I don't know if this is just a thing with how GitHub displays the diff, but the bullet points here aren't in the same order as the pages that talk about these subjects, I think it'd be nice to align them.

Reply:
The order they are in is specified in the toc.json:

```json
{
  "npm_nodejs": {
    "name": "Using npm/Node.js code",
    "children": {
      "std_node": "The std/node library",
      "cdns": "Packages from CDNs",
      "import_maps": "Using import maps",
      "faqs": "Frequently asked questions"
    }
  }
}
```

And the order here matches. 😕

Reply:
Ah great, didn't know about that part :+1: